gapic/src/main/com/google/gapid/perfetto/models/PowerSummaryTrack.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2022 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gapid.perfetto.models;
+
+import static com.google.gapid.perfetto.models.QueryEngine.createSpan;
+import static com.google.gapid.perfetto.models.QueryEngine.createView;
+import static com.google.gapid.perfetto.models.QueryEngine.createWindow;
+import static com.google.gapid.perfetto.models.QueryEngine.dropTable;
+import static com.google.gapid.perfetto.models.QueryEngine.dropView;
+import static com.google.gapid.perfetto.views.StyleConstants.POWER_RAIL_COUNTER_TRACK_HEIGHT;
+import static com.google.gapid.perfetto.views.TrackContainer.group;
+import static com.google.gapid.perfetto.views.TrackContainer.single;
+import static com.google.gapid.util.MoreFutures.transform;
+import static com.google.gapid.util.MoreFutures.transformAsync;
+import static java.lang.String.format;
+
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.gapid.models.Perfetto;
+import com.google.gapid.perfetto.Unit;
+import com.google.gapid.perfetto.models.TrackConfig.Group;
+import com.google.gapid.perfetto.views.CounterPanel;
+import com.google.gapid.perfetto.views.PowerSummaryPanel;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** {@link Track} summarizing the total Power usage. */
+public class PowerSummaryTrack extends Track.WithQueryEngine<PowerSummaryTrack.Data> {
+  private static String viewSql;
+  private static int numPowerRailTracks;
+  public static Unit unit;
+  public static Double minValue;
+  public static Double maxValue;
```

Comment:
Avoid mutable static state. This would break if multiple traces are loaded. These fields are only used from an instance, so just make them instance fields.

Reply:
Makes sense, updated.